app/cli/internal/policydevel/lint.go

```diff
@@ -354,38 +356,77 @@ func (p *PolicyToLint) loadDefaultConfig() (*config.Config, error) {
 	return &cfg, nil
 }
 
-// Splits grouped errors into individual errors
-func (p *PolicyToLint) processRegalViolation(rawErr error, path string) {
-	if rawErr == nil {
+// Formats regal violation errors
+func (p *PolicyToLint) processRegalViolation(errorStr, path string) {
```

Comment:
I've been tying this locally, and I've found that you already have all the details in the `err` object coming from opa. Check this out https://github.com/open-policy-agent/opa/blob/main/v1/ast/errors.go, which is the type of the `err` object, at least the one returned by `InputFromText`.. You can iterate through all errors and get what you need without building nor parsing any string.